src/main/java/edu/wpi/grip/ui/pipeline/input/NumberSpinnerInputSocketView.java

```diff
@@ -0,0 +1,66 @@
+package edu.wpi.grip.ui.pipeline.input;
+
+import com.google.common.eventbus.EventBus;
+import com.google.common.eventbus.Subscribe;
+import edu.wpi.grip.core.InputSocket;
+import edu.wpi.grip.core.events.SocketChangedEvent;
+import javafx.beans.InvalidationListener;
+import javafx.scene.control.Spinner;
+import javafx.scene.control.SpinnerValueFactory;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+/**
+ * An {@link InputSocketView} that lets the user set the value of a number with a {@link Spinner}
+ */
+public class NumberSpinnerInputSocketView extends InputSocketView<Number> {
+
+    private final static Object[] DEFAULT_DOMAIN = new Object[]{Double.MIN_VALUE, Double.MAX_VALUE};
+
+    private final SpinnerValueFactory<Double> valueFactory;
+    private final InvalidationListener updateSocketFromSpinner;
+
+    /**
+     * @param socket An <code>InputSocket</code> with a domain containing two <code>Number</code>s (the min and max
+     *               slider values), or no domain at all.
+     */
+    public NumberSpinnerInputSocketView(EventBus eventBus, InputSocket<Number> socket) {
+        super(eventBus, socket);
+
+
+        Object[] domain = socket.getSocketHint().getDomain();
+        if (domain == null) {
+            domain = DEFAULT_DOMAIN;
+        }
+
+        checkArgument(domain.length == 2 && domain[0] instanceof Number && domain[1] instanceof Number,
+                "Spinners must have a domain with two numbers (min and max)");
```

Comment:
I want to preface this with a "I'm not adamantly against this pull request, I'm looking to have a discussion about how to do this right." As to your above points: 
1. I agree.
2. Whatever
3. I would prefer to see a `public void validate()` method on SocketHints. Not copy and paste.
4. A SocketHint is invalid if it prevents the creation of an step in the pipeline. This would prevent a step from being created.
5. I'm not arguing to keep validation in the InputSocketView.